### PR TITLE
build improvements

### PR DESCRIPTION
**Pull request by reflectored**
## Description

Enable Linux build via a Dockerfile (Ubuntu), add compile commands generation for clangd LSP and clean up .gitignore.

### CODE REVIEW

**Comment by m-kuhn** [comment]
There are already some Docker files in the `.docker` folder, I think this one here should be maintained outside the QGIS repository.
For the other changes I'm unsure what they bring/why they are needed.

File: docker_build/Dockerfile
Comment (troopa81) [inline docker_build/Dockerfile:1]: there is already docker files in the .docker folder

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@ set(CPACK_PACKAGE_VERSION_PATCH "0")
 set(COMPLETE_VERSION ${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH})
 set(RELEASE_NAME "Master")
 project(qgis VERSION ${COMPLETE_VERSION})
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (troopa81) [inline CMakeLists.txt:25]: I'm using LSP, and it's better to add this option at command line so it doesn't bother other people building QGIS without LSP

File: .gitignore [hunk 2]
```diff
@@ -39,6 +39,8 @@ i18n/*.qm
 Makefile
 *.pro.user
 *.stash
+.ccache/
+.ci/
```
Comment (troopa81) [inline .gitignore:43]: actually .ci cannot be ignored, it's part of the QGIS ci configuration

File: src/core/CMakeLists.txt [hunk 3]
```diff
@@ -2135,7 +2135,7 @@ set(IMAGE_RCCS ../../images/images.qrc)
 
 #############################################################
 # qgis_core library
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (troopa81) [inline src/core/CMakeLists.txt:2138]: Why ? we need binary directory for generated header IIRC

**Review by troopa81 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (reflectored) [reply .gitignore:43]: ack.

File: docker_build/Dockerfile
Comment (reflectored) [reply docker_build/Dockerfile:1]: The build works with the official QGIS docker image, thanks!

**Pull request closed without merging**